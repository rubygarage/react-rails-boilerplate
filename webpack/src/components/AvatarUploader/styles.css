.avatar-preview-wrap {
  height: 100%;
  width: 100%;
  position: relative;
}

.avatar-uploader-thumb {
  height: 100%;
  width: 100%;
  object-fit: cover;
}

.remove-avatar-preview-button {
  position:absolute;
  top: 5px;
  right: 5px;
  font-size: 13px;
  height: 20px;
  width: 20px;
  border-radius: 10px;
  border: none;
  color: #ffffff;
  background-color: #333333;
  cursor: pointer;
}
