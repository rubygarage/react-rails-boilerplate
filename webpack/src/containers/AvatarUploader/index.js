import React, { Component } from 'react';
import PropTypes from 'prop-types';
import AvatarUploaderComponent from 'components/AvatarUploader';

class AvatarUploader extends Component {
  static propTypes = {
    input: PropTypes.shape({
      value: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.array,
      ]),
      onChange: PropTypes.func,
    }),
  };

  fileDropHandler = (accepted) => {
    if (accepted && accepted.length) { this.props.input.onChange(accepted); }
  };

  handleRemoveSelectedFile = () => {
    this.props.input.onChange('');
  };

  render() {
    const props = {
      ...this.props,
      handleRemoveSelectedFile: this.handleRemoveSelectedFile,
      fileDropHandler: this.fileDropHandler,
    };

    return <AvatarUploaderComponent {...props} />;
  }
}

export default AvatarUploader;
