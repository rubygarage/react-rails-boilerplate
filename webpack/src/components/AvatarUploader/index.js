import React from 'react';
import PropTypes from 'prop-types';
import Dropzone from 'react-dropzone';

import styles from './styles.css';

function AvatarUploader({
  input, name, multiple, handleRemoveSelectedFile, fileDropHandler,
}) {
  const files = input.value;
  return (
    <div>
      <Dropzone
        multiple={multiple}
        name={name}
        accept="image/jpeg, image/png"
        onDrop={fileDropHandler}
      >
        { files && Array.isArray(files)
          ?
            files.map(file => (
              <div className={styles['avatar-preview-wrap']} key={file.name}>
                <img src={file.preview} alt="Uploaded avatar" className={styles['avatar-uploader-thumb']} />
                <button
                  type="button"
                  className={styles['remove-avatar-preview-button']}
                  onClick={handleRemoveSelectedFile}
                >
                  x
                </button>
              </div>
            ))
          :
            <div>Try dropping some files here, or click to select files to upload.</div>
        }
      </Dropzone>
    </div>
  );
}

AvatarUploader.propTypes = {
  input: PropTypes.shape({
    value: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.array,
    ]),
    onChange: PropTypes.func,
  }),
  handleRemoveSelectedFile: PropTypes.func.isRequired,
  fileDropHandler: PropTypes.func,
  multiple: PropTypes.bool,
  name: PropTypes.string,
};

export default AvatarUploader;
