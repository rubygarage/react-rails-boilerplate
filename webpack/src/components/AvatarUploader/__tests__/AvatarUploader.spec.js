import React from 'react';
import { shallow } from 'enzyme';
import AvatarUploader from 'components/AvatarUploader';

describe('Show user', () => {
  let props = {
    input: {
      value: '',
      onChange: jest.fn(),
    },
    handleRemoveSelectedFile: jest.fn(),
    fileDropHandler: jest.fn(),
    multiple: false,
    name: 'avatar',
  };

  it('renders no preview when no file is selected', () => {
    const component = shallow(<AvatarUploader {...props} />);

    expect(component).toMatchSnapshot();
  });

  it('renders preview when file is selected', () => {
    props = {
      ...props,
      input: {
        value: [
          {
            name: 'filename',
            preview: 'filepreview.jpg',
          },
        ],
      },
    };
    const component = shallow(<AvatarUploader {...props} />);

    expect(component).toMatchSnapshot();
  });
});
