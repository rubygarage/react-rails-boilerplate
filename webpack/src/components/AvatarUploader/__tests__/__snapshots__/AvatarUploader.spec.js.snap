// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show user renders no preview when no file is selected 1`] = `
<div>
  <t
    accept="image/jpeg, image/png"
    disableClick={false}
    disablePreview={false}
    disabled={false}
    maxSize={Infinity}
    minSize={0}
    multiple={false}
    name="avatar"
    onDrop={[Function]}
    preventDropOnDocument={true}
  >
    <div>
      Try dropping some files here, or click to select files to upload.
    </div>
  </t>
</div>
`;

exports[`Show user renders preview when file is selected 1`] = `
<div>
  <t
    accept="image/jpeg, image/png"
    disableClick={false}
    disablePreview={false}
    disabled={false}
    maxSize={Infinity}
    minSize={0}
    multiple={false}
    name="avatar"
    onDrop={[Function]}
    preventDropOnDocument={true}
  >
    <div
      className="avatar-preview-wrap"
      key="filename"
    >
      <img
        alt="Uploaded avatar"
        className="avatar-uploader-thumb"
        src="filepreview.jpg"
      />
      <button
        className="remove-avatar-preview-button"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
  </t>
</div>
`;
